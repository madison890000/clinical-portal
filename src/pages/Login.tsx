import { useContext, useEffect } from 'react';
import { Typography } from '@mui/material';
import { AppContext } from '../contexts/AppContext';
import { useNavigate } from 'react-router';
import styles from './Login.module.scss';
import { LoginStatus } from '../types';
import { PORTAL_NAME, ROUTES } from '../constants';
import Footer from '../modules/Footer';
import LoginForm from '../modules/LoginForm';

type Inputs = {
    username: string;
    password: string;
};
const Login = () => {
    const { login, loginStatus } = useContext(AppContext);
    const navigate = useNavigate();
    useEffect(() => {
        if (loginStatus === LoginStatus.Login) {
            navigate(ROUTES.ClinicianDetail);
        }
    }, [loginStatus, navigate]);
    const onSubmit = async ({ username, password }: Inputs) => {
        await login(username, password);
        navigate('/clinician-detail');
    };
    return (
        <>
            <div className={styles.title}>
                <Typography variant="h5" align="center">
                    {PORTAL_NAME} Sign In
                </Typography>
            </div>
            <div className={styles.loginContainer}>
                <LoginForm onSubmit={onSubmit} />
            </div>
            <Footer />
        </>
    );
};

export default Login;
