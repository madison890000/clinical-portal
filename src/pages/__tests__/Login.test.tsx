import React from 'react';
import Login from '../Login';
import { render, waitFor } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

jest.mock('../../services', () => {
    return {
        getPatientById: () => {
            return Promise.resolve({
                title: 'Mr.',
                firstName: 'James',
                preferredName: 'Jay',
                middleName: 'Shaun',
                familyName: 'Cooper',
                age: 17,
                sex: 'Indeterminate'
            });
        }
    };
});
test('renders Login Page', async () => {
    const { container } = render(
        <BrowserRouter>
            <Login />
        </BrowserRouter>
    );
    await waitFor(() => {
        expect(container).toMatchSnapshot();
    });
});
