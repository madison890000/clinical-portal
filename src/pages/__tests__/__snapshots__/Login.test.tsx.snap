// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login Page 1`] = `
<div>
  <div
    class="title"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-ph2xsj-MuiTypography-root"
    >
      Clinical Portal
       Sign In
    </h5>
  </div>
  <div
    class="loginContainer"
  >
    <form>
      <p
        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
      >
        Username or email address
      </p>
      <div
        class="inputContainer"
      >
        <div
          class="MuiFormControl-root MuiTextField-root input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r0:"
            id=":r0:-label"
          >
            Required*
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="sign-in-username"
              id=":r0:"
              name="username"
              placeholder="please input username"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Required*
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="passwordContainer"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          Password
        </p>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
        >
          Forgot password?
        </a>
      </div>
      <div
        class="inputContainer"
      >
        <div
          class="MuiFormControl-root MuiTextField-root input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1pysi21-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for=":r1:"
            id=":r1:-label"
          >
            Required*
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
              data-testid="sign-in-password"
              id=":r1:"
              name="password"
              placeholder="please input password"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-yjsfm1"
              >
                <span>
                  Required*
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <div
        class="buttonContainer"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiLoadingButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium signInContainer css-1twrztz-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
          data-testid="sign-in-btn"
          id=":r2:"
          tabindex="0"
          type="submit"
        >
          Sign in
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </form>
  </div>
  <div
    class="footer"
  >
    <div>
      Terms
    </div>
    <div>
      Privacy
    </div>
    <div>
      Security
    </div>
    <div>
      Contact us
    </div>
  </div>
</div>
`;
